src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingQueriesSpec.cs

```diff
@@ -0,0 +1,193 @@
+        {
+            AwaitClusterUp(config.Controller, config.Busy, config.Second, config.Third);
+
+            RunOn(() =>
+            {
+                StartProxy(
+                    Sys,
+                    typeName: ShardTypeName,
+                    role: "shard",
+                    extractEntityId: extractEntityId,
+                    extractShardId: extractShardId);
+            }, config.Controller);
+
+            RunOn(() =>
+            {
+                StartSharding(
+                    Sys,
+                    typeName: ShardTypeName,
+                    entityProps: Props.Create(() => new PingPongActor()),
+                    settings: settings.Value.WithRole("shard"),
+                    extractEntityId: extractEntityId,
+                    extractShardId: extractShardId);
+            }, config.Busy, config.Second, config.Third);
+
+            EnterBarrier("sharding started");
+        }
+
+        private void Querying_cluster_sharding_must_trigger_sharded_actors()
+        {
+            RunOn(() =>
+            {
+                Within(TimeSpan.FromSeconds(10), () =>
+                {
+                    AwaitAssert(() =>
+                    {
+                        var pingProbe = CreateTestProbe();
+                        foreach (var n in Enumerable.Range(0, 20))
+                        {
+                            _region.Value.Tell(new PingPongActor.Ping(n), pingProbe.Ref);
+                        }
+                        pingProbe.ReceiveWhile(null, m => (PingPongActor.Pong)m, 20);
+                    });
+                });
+            }, config.Controller);
+            EnterBarrier("sharded actors started");
+        }
+
+        private void Querying_cluster_sharding_must_return_shard_stats_of_cluster_sharding_regions_if_one_or_more_shards_timeout_versus_all_as_empty()
+        {
+            RunOn(() =>
+            {
+                var probe = CreateTestProbe();
+                var region = ClusterSharding.Get(Sys).ShardRegion(ShardTypeName);
+                region.Tell(new GetClusterShardingStats(TimeSpan.FromSeconds(10)), probe.Ref);
+                var regions = probe.ExpectMsg<ClusterShardingStats>().Regions;
+                regions.Count.Should().Be(3);
+                var timeouts = NumberOfShards / regions.Count;
+
+                // 3 regions, 2 shards per region, all 2 shards/region were unresponsive
+                // within shard-region-query-timeout, which only on first is 0ms
+                regions.Values.Select(i => i.Stats.Count).Sum().Should().Be(4);
+                regions.Values.Select(i => i.Failed.Count).Sum().Should().Be(timeouts);
+            }, config.Busy, config.Second, config.Third);
+            EnterBarrier("received failed stats from timed out shards vs empty");
+        }
+
+        private void Querying_cluster_sharding_must_return_shard_state_of_sharding_regions_if_one_or_more_shards_timeout_versus_all_as_empty()
```

Comment:
Nice - I've run into this issue before with `pbm cluster-sharding region stats -r {regionName}` before; glad this has been fixed.